tools/udhcpc-override.sh

```diff
@@ -87,10 +87,23 @@ fi
     UDHCPC_OPTS="$UDHCPC_OPTS_DEFAULT"
 
 case "$UDHCPC_OPTS" in
-    *hostname*) ;;
+    *hostname*) ;; # Something provided already
     *)  # Give hint about my non-default name
-        [ x"`hostname`" != xeaton-rc3 ] && \
-        UDHCPC_OPTS="$UDHCPC_OPTS -x hostname:`hostname`"
+        case x"`hostname`" in
+            x|xeaton-rc3|xlocalhost*)
+                echo "WARNING: Current local host name is '`hostname`', trying to find a better name" >&2
+                # Try to generate and apply a MAC-based name, do not save it
+                # yet - might do so through DHCP assignment processing though
+                interface="$UDHCPC_IFACE" \
+                    fty-hostname-setup "" "false"
+            ;;
+        esac
+
+        case x"`hostname`" in
+            x|xeaton-rc3|xlocalhost*)
+                echo "WARNING: Current local host name is '`hostname`', so not pushing it to DHCP" >&2 ;;
+            *) UDHCPC_OPTS="$UDHCPC_OPTS -x hostname:`hostname`" ;;
```

Comment:
Just for clarification, is this still valid in case of empty hostname?

Reply:
You're right, I missed the *) on the next line, now it makes much more sense, sorry